client/app/templates/template-editor.html

```diff
@@ -0,0 +1,75 @@
+    <div class="panel-body">
+      <section>
+        <div class="col-md-12 ss-details-header">
+          <div class="row">
+            <div class="col-lg-7 col-md-6 col-sm-6 col-xs-6">
+              <h2 translate ng-if="vm.template.id">Edit Template</h2>
+              <h2 translate ng-if="!vm.template.id">Add Template</h2>
+            </div>
+          </div>
+        </div>
+      </section>
+      <section class="ss-form">
+        <form  name="templateForm" class="form-horizontal">
+        <div class="col-md-12">
+          <div class="row">
+            <div class="col-lg-7 col-md-6 col-sm-6 col-xs-6">
+              <h3 translate>Basic Information</h2>
+            </div>
+          </div>
+          <div class="row">
+            <div class="col-lg-7 col-md-7 col-sm-7 col-xs-7">
+              <div class="form-horizontal">
+                <div class="form-group">
+                  <label class="control-label col-sm-4" translate>Name *</label>
+                  <div class="col-sm-8">
+                    <input id="template-editor-name"
+                           class="form-control"
+                           ng-model="vm.template.name"/>
+                  </div>
+                </div>
+                <div class="form-group">
+                  <label class="control-label col-sm-4">{{'Description' | translate}}</label>
+                  <div class="col-sm-8">
+                       <textarea class="form-control" rows="5" ng-model="vm.template.description" />
+                  </div>
+                </div>
+                <div class="form-group test">
+                  <label class="control-label col-sm-4">{{'Template Type' | translate}}</label>
+                  <div class="col-sm-8">
+                  <select-dropdown options="vm.templateTypes" model="vm.templateTypeValue" id="templateType"></select-dropdown>
+                  </div>
+                </div>
+                <div>
+                  <label class="control-label col-sm-4">{{'Draft' | translate}}</label>
+                  <div class="col-sm-8">
+                      <input type="checkbox" ng-model="vm.template.draft" ng-true-value="true" ng-false-value="false" />
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+          <div class="row">
+            <div class="col-lg-7 col-md-6 col-sm-6 col-xs-6">
+              <h3 translate>Code</h2>
+            </div>
+          </div>
+          <div ace-editor content="vm.template.content" class="template-content"></div>
+        </div>
+        </form>
+      </section>
+    </div>
+  </div>
+  <div>
+    <div class="panel panel-default ss-details-panel ss-details-panel-bottom">
+      <div class="panel-body">
+        <section class="ss-details-section pull-left">
+          <button class="btn btn-primary" type="button" ng-click="vm.saveTemplate()" ng-disabled="!vm.isPristine()" translate>Save</button>
```

Comment:
ng-disabled should be something along the lines of `templateForm.$pristine` no need for a function here